packages/queries/src/types/ColumnValue.ts

```diff
@@ -14,15 +14,11 @@ export type ColumnValue<
     DB,
     TTable extends Table<DB>,
     TColumn extends Column<DB, TTable>,
-> =
-    // Case 1: The value is a ColumnType
-    DB[TTable][TColumn] extends Selectable<infer T>
-        ? T
-        : // Case 2: The value is a nullable ColumnType
-          DB[TTable][TColumn] extends Selectable<infer T> | null
-          ? T | null
-          : DB[TTable][TColumn];
+> = DB[TTable] extends { columns: infer TColumnDef }
+    ? ColumnType<TColumnDef, TColumn>
+    : never;
 
-type Selectable<T> = {
-    readonly __select__: T;
-};
+type ColumnType<
+    TColumnDef,
+    TColumn extends keyof TColumnDef,
+> = TColumnDef[TColumn] extends { type: infer TType } ? TType : never;
```

Comment:
There's likely a bug here: this isn't taking nullable columns into account. Should be something like

Case 1:
{type: infer TType, isNullable: true} ? null | TType

Case 2:
{type: infer TType, isNullable:false } ? TType

Case 3:
never